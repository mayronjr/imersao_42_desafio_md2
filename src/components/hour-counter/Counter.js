import 'bootstrap/dist/css/bootstrap.min.css';

function Counter({counter}) {
    let text = counter + " hours of exercises"
    return (
        <div className="container" data-testid="counter">
            {text}
        </div>
    );
}

export default Counter;
