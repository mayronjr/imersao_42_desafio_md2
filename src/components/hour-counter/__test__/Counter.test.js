import React from 'react'
import ReactDOM from 'react-dom'
import Counter from '../Counter.js'

import { render, cleanup } from '@testing-library/react'
import "@testing-library/jest-dom/extend-expect"

import renderer from 'react-test-renderer'

afterEach(cleanup)

it('renders without crashing', ()=>{
    const div = document.createElement('div')
    ReactDOM.render(<Counter/>, div)
})

it('renders Counter Correctly', ()=>{
    const {getByTestId} = render(<Counter counter={1}/>)
    expect(getByTestId('counter')).toHaveTextContent("1 hours of exercises")
})
it('renders Counter Correctly', ()=>{
    const {getByTestId} = render(<Counter counter={5}/>)
    expect(getByTestId('counter')).toHaveTextContent("5 hours of exercises")
})

it('matches snapshot', ()=>{
    const tree = renderer.create(<Counter counter={3}/>).toJSON()
    expect(tree).toMatchSnapshot()
})