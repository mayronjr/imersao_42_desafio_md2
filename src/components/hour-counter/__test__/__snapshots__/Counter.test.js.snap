// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container"
  data-testid="counter"
>
  3 hours of exercises
</div>
`;
